### PR TITLE
Refactor payment-related blocks registration

### PR DESCRIPTION
**Pull request by monsieur-z**
- See https://github.com/Automattic/jetpack/issues/32602
- See https://github.com/Automattic/jetpack/issues/33174

## Proposed changes:

This PR is a follow up of https://github.com/Automattic/jetpack/pull/33842, which was reverted in https://github.com/Automattic/jetpack/pull/33895. The differences between the 2 PRs are the following:
- The simple payments block API version was set to 1 (this was done in https://github.com/Automattic/jetpack/pull/33889).
- The default value of the `currency` attribute for the same block was set to `USD` (replacing an invalid value).

See p1698765346173339-slack-C052XEUUBL4 for more details.

_Below's the description of the original PR._

Registers the following production blocks by specifying the path to their block.json file:  _Payments Intro_, _Paywall_, _Premium Content_, _Recurring Payment_, _Simple Payment_, _Subscriptions_.

This was made possible by https://github.com/Automattic/jetpack/pull/32697. The goal is to avoid the duplication of block metadata and use block.json as the source of truth.

In more details, for each block (when needed):
  - Write reading direction-agnostic CSS
  - Remove the `editorScript` property from `block.json` (automatically added during the build)
  - Convert icon from dashicon to SVG in the `icon` property of `block.json`
  - Remove `icon.js`
  - Import icon helpers from the `shared-extension-utils` package
  - Register the block client and server sides with `block.json`
  - Move data from `index.js` to `block.json` and `editor.js` and delete the file
  - Move data from `attributes.js` to `block.json` and delete the file
  - Fix JS tests

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pedMtX-RS-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Spin up a test site
- Create a new post and open the block inserter
- Test that the aforementioned blocks behave as in `trunk`. This PR shouldn't change anything from a user's standpoint.
- Optionally, test these blocks on a simple site

### CODE REVIEW

**Comment by anomiex** [comment]
LGTM, the issue with the payment block is fixed now in some manual tests. Remind me to approve this after 12.8 is cut, it's too close right now for me to feel comfortable trying this again now.

**Comment by anomiex** [comment]
Ok, 12.8 is out. This needs a trunk merge now, then we can give it a try.

**Review by anomiex — DISMISSED** [review]

**Review by jeherve — APPROVED** [review]
This tests well for me. 🚢

File: projects/plugins/jetpack/extensions/blocks/paywall/editor.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import { unregisterBlockType } from '@wordpress/blocks';
 import { subscribe, select } from '@wordpress/data';
 import { store as editorStore } from '@wordpress/editor';
-import registerJetpackBlock from '../../shared/register-jetpack-block';
-import { name, settings } from '.';
```
Comment (fluiddot) [inline projects/plugins/jetpack/extensions/blocks/paywall/editor.js:5]: 👋 @monsieur-z, I noticed that `name` is being used in line 21:
https://github.com/Automattic/jetpack/blob/30f477439a16f9720b63523b64a314846e02483a/projects/plugins/jetpack/extensions/blocks/paywall/editor.js#L21

This produces a crash in the Jetpack app. Could we revert this change and keep the `name` property?
Thanks 🙇 !

Comment (fluiddot) [reply projects/plugins/jetpack/extensions/blocks/paywall/editor.js:5]: I went ahead and created a PR to fix this: https://github.com/Automattic/jetpack/pull/34274.
Let me know if you could take a quick look, thanks 🙇  !
